cv_next/app/cv/[cvId]/components/commentSection/comment.tsx

```diff
@@ -28,69 +29,71 @@ export default function Comment({
   commentsOfComment = [],
   setCommentsOfComments,
 }: {
-  comment: CommentModel
-  userId: string
-  childDepth?: number
-  cv: CvModel
-  commentsOfComment: any[]
+  comment: CommentModel;
+  userId: string;
+  childDepth?: number;
+  cv: CvModel;
+  commentsOfComment: any[];
   setCommentsOfComments: (
     update: (prev: Map<string, any[]>) => Map<string, any[]>
-  ) => void
+  ) => void;
 }) {
   const [commentOnCommentStatus, setCommentOnCommentStatus] =
-    useState<boolean>(false)
-  const [commentOnComment, setCommentOnComment] = useState<string>("")
-  const [submitCommentOnComment, setSubmitCommentOnComment] =
-    useState<boolean>(false)
-
-  const [showAlert, setShowAlert] = useState<boolean>(false)
+    useState<boolean>(false);
+  const [commentOnComment, setCommentOnComment] = useState<string>("");
+  const router = useRouter();
+  const [showAlert, setShowAlert] = useState<boolean>(false);
   const onAlertClick = async (type: boolean) => {
-    if (type) await deleteCommentAction()
-    setShowAlert(false)
-  }
+    if (type) await deleteCommentAction();
+    setShowAlert(false);
+  };
+  const supabase = useSupabase();
 
   const addNewCommentClickEvent = async () => {
-    if (!userId) throw new Error("User not found")
-    const commentToAdd: NewCommentModel = {
-      data: commentOnComment,
-      document_id: cv.id,
-      parent_comment_Id: comment.id,
-      user_id: userId,
-    }
+    const userId = (await supabase.auth.getUser()).data.user?.id;
+    if (!userId) router.push("/inactive");
+    else {
+      const commentToAdd: NewCommentModel = {
+        data: commentOnComment,
+        document_id: cv.id,
+        parent_comment_Id: comment.id,
+        user_id: userId,
+      };
 
-    await addNewComment(commentToAdd).finally(() => {
-      setCommentsOfComments((prev: Map<string, any[]>) => {
-        const newCommentsOfComments = new Map<string, any[]>(prev)
-        newCommentsOfComments.set(
-          comment.id,
-          newCommentsOfComments
-            .get(comment.id)
-            ?.concat([commentToAdd]) as Array<any>
-        )
-        return newCommentsOfComments
-      })
-      setCommentOnCommentStatus(false)
-    })
-    mutate(comment.document_id)
-  }
+      await addComment(commentToAdd).finally(() => {
```

Comment:
Instead of the `newCommentsOfComments` (ofCommentsOfCommentsOfComments....) fiasco, we can try utilizing swr's [optimistic update](https://swr.vercel.app/docs/mutation#optimistic-updates) and rollback if needed.